src/components/select/bl-select.ts

```diff
@@ -0,0 +1,394 @@
+              maxHeight: `${
+                args.availableHeight > defaultMaxHeight ? defaultMaxHeight : args.availableHeight
+              }px`,
+            });
+          },
+        }),
+      ],
+    }).then(({ x, y }) => {
+      this._selectMenu.style.setProperty('--left', `${x}px`);
+      this._selectMenu.style.setProperty('--top', `${y}px`);
+    });
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    document.addEventListener('click', this.clickOutsideHandler);
+  }
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    document.removeEventListener('click', this.clickOutsideHandler);
+  }
+
+  inputTemplate() {
+    const inputSelectedOptions = html`<ul class="selected-options">
+      ${this._selectedOptions.map(item => html`<li>${item.text}</li>`)}
+    </ul>`;
+    const _selectedItemCount = this._additionalSelectedOptionCount
+      ? html`<span>+${this._additionalSelectedOptionCount}</span>`
+      : null;
+    const removeIcon = this._isRemoveIconShow
+      ? html`<bl-icon
+          class="remove-all"
+          name="close"
+          title="close"
+          style="font-size: var(--bl-font-size-xs);"
+          @click=${this._onClickRemove}
+        ></bl-icon>`
+      : null;
+
+    const placeholder = this._showPlaceHolder
+      ? html`<span class="placeholder">${this.placeholder}</span>`
+      : '';
+
+    return html`<div
+      class="select-input"
+      ?disabled=${this.disabled}
+      @click=${this._onClickSelectInput}
+    >
+      ${placeholder} ${inputSelectedOptions} ${_selectedItemCount}
+      <div class="actions">
+        ${removeIcon}
+        <bl-icon
+          name="${this._isMenuOpen ? 'arrow_up' : 'arrow_down'}"
+          title="arrow_down"
+          style="font-size: var(--bl-font-size-m)"
+        ></bl-icon>
+      </div>
+    </div>`;
+  }
+
+  menuTemplate() {
+    return html`<div class="select-menu" @bl-select-option=${this._handleSelectOptionEvent}>
+      <slot></slot>
+    </div>`;
+  }
+
+  render() {
+    const invalidMessage =
+      this._isInvalid && this.customInvalidText
+        ? html`<p class="invalid-text">${this.customInvalidText}</p>`
+        : ``;
+    const helpMessage =
+      this.helpText && !invalidMessage ? html`<p class="help-text">${this.helpText}</p>` : ``;
+    const label = this.label ? html`<label>${this.label}</label>` : '';
+
+    return html`<div
+      class=${classMap({
+        'select-wrapper': true,
+        'select-open': this._isMenuOpen,
+        'selected': this._selectedOptions.length > 0,
+        'invalid': this._isInvalid,
+      })}
+      tabindex="-1"
+    >
+      ${label} ${this.inputTemplate()} ${this.menuTemplate()} ${helpMessage} ${invalidMessage}
+    </div> `;
+  }
+
+  private get _showPlaceHolder() {
+    if (this.label && !this.labelFixed) {
+      return !this._selectedOptions.length && this._isMenuOpen;
+    }
+    return !this._selectedOptions.length;
+  }
+
+  private get _isRemoveIconShow() {
```

Comment:
I suggest to use a different name. Like `isRemoveIconVisible` or `showRemoveIcon`